thunder/tests/test_examine_memory.py

```diff
@@ -150,7 +150,7 @@ def test_nanogpt_block(executor, device, dtype):
         expected_return_calculated_mem = get_return_memory(fw_extrace.bound_symbols[-1]) - 4 * 2 * 1024 * 768
         assert expected_return_calculated_mem == sum(fw_alloc_mem[1].values())
 
-        assert bw_alloc_mem[0] == 361783296
+        assert bw_alloc_mem[0] == 361881600
```

Comment:
This change is necessary because we are rejecting sum here, since it's taking the symbolic proxy as reduction axes and being rejected by nvfuser.

Reply:
That's a little concerning. What else might we be kicking out of an nvFuser region that previously got inserted?

Your experience here makes me think the old behavior (which AFAIK is baking the value into the graph as a constant and recompiling the graph should the value change) should be the default, and we should use a compile option to turn this on. At least for a little while.

Reply:
> Your experience here makes me think the old behavior (which AFAIK is baking the value into the graph as a constant and recompiling the graph should the value change) should be the default

That's a reasonable argument. We certainly can leave that as the default.

> and we should use a compile option to turn this on. At least for a little while.

This is effectively done today with `cache` option in `thunder.jit`.

> What else might we be kicking out of an nvFuser region that previously got inserted?

I would argue that the updated nvFuser behavior is the right thing to do. i.e. if we see a NumberProxy on a place where we cannot handle it, we should kick it out. You could argue that nvfuser should have updated its support on that and that's a conversation I'm happy to continue.

Reply:
> I would argue that the updated nvFuser behavior is the right thing to do.

Sorry, I didn't mean to imply that it wasn't!

But I also want to make sure we're not going to see a bunch of perf regressions after this PR. Maybe that's not a real concern? If it is, an alternate idea might be to update the lit-gpt benchmarks to set the cache option to the old behavior

Reply:
> But I also want to make sure we're not going to see a bunch of perf regressions after this PR. Maybe that's not a real concern?

Right now `NumberProxy` requires explicit `opt-in` via specifying cache policy to `symbolic values`. We are not changing the default behavior so we should be fine.

One exception is that grad transform (as well as other transforms) might introduce `NumberProxy`, but it's not doing that for reduction axes right now. https://github.com/Lightning-AI/lightning-thunder/blob/7ac4c49aca1f4282dc9dfe5ba69e343a358d1942/thunder/core/transforms.py#L1168-L1178

I'll be surprised if reduction axes are given as runtime inputs. Looks at the `_normalize` in `torch/__init__.py`, which only uses tensor rank to compute the reduction axes  https://github.com/Lightning-AI/lightning-thunder/blob/7ac4c49aca1f4282dc9dfe5ba69e343a358d1942/thunder/torch/__init__.py#L2872-L2873

Reply:
Ahh, I had misunderstood then. sounds great, thank you!